src/Microsoft.AspNet.Mvc.ModelBinding/Binders/CollectionModelBinder.cs

```diff
@@ -133,13 +165,49 @@ public class CollectionModelBinder<TElement> : IModelBinder
             return boundCollection;
         }
 
-        // Extensibility point that allows the bound collection to be manipulated or transformed before
-        // being returned from the binder.
+        /// <summary>
+        /// Gets an <see cref="object"/> suitable for the associated property.
+        /// </summary>
+        /// <param name="newCollection">
+        /// Collection of values retrieved from value providers. Or <c>null</c> if nothing was bound.
+        /// </param>
+        /// <returns>
+        /// <see cref="object"/> suitable for the associated property. Or <c>null</c> if nothing was bound.
+        /// </returns>
+        /// <remarks>
+        /// Extensibility point that allows the bound collection to be manipulated or transformed before being
+        /// returned from the binder.
+        /// </remarks>
         protected virtual object GetModel(IEnumerable<TElement> newCollection)
         {
+            // Depends on fact BindSimpleCollection() and BindComplexCollection() always return a List<TElement>
+            // instance or null. In addition GenericModelBinder confirms a List<TElement> is assignable to the
+            // property prior to instantiating this binder and subclass binders do not call this method.
             return newCollection;
         }
 
+        /// <summary>
+        /// Adds values from <paramref name="sourceCollection"/> to given <paramref name="target"/>.
+        /// </summary>
+        /// <param name="target"><see cref="object"/> into which values are copied.</param>
+        /// <param name="sourceCollection">
+        /// Collection of values retrieved from value providers. Or <c>null</c> if nothing was bound.
+        /// </param>
+        /// <remarks>Called only in TryUpdateModelAsync(collection, ...) scenarios.</remarks>
+        protected virtual void CopyToModel([NotNull] object target, IEnumerable<TElement> sourceCollection)
```

Comment:
this part of the feature was removed (along w/ `ModelValidationNode`) in f19c2e4.  previous `CollectionModelBinder.CreateOrReplaceCollection()` method checked equivalent of `targetCollection.IsReadOnly` called `targetCollection.Clear()` before doing the additions.  restore both in next iteration...